eulerpi/examples/heat/heat.py

```diff
@@ -0,0 +1,210 @@
+
+        solution = self.perform_simulation(self, param)
+        # return the solution at four evaluation points
+        eval_points = jnp.multiply(
+            jnp.array(
+                [
+                    [0.25, 0.25, 0.25],
+                    [0.25, 0.75, 0.25],
+                    [0.25, 0.25, 0.75],
+                    [0.25, 0.75, 0.75],
+                ]
+            ),
+            jnp.array(
+                [solution.shape[0], solution.shape[1], solution.shape[2]]
+            ),
+        ).astype(int)
+        sim_res = jnp.array(
+            solution[eval_points[:, 0], eval_points[:, 1], eval_points[:, 2]]
+        )
+        return sim_res
+
+    def param_is_within_domain(self, param: np.ndarray) -> bool:
+        """Checks whether a parameter is within the parameter domain of the model.
+        This condition stems from thermodynamical considerations.
+
+        Args:
+            param(np.ndarray): The parameter to check.
+
+        Returns:
+            bool: True if the parameter is within the limits.
+
+        """
+        return param[0] * param[1] > param[2] ** 2
+
+    def perform_simulation(self, param: np.ndarray) -> np.ndarray:
+        """Performs a simulation of the heat equation with the given parameters.
+
+        Args:
+            param (np.ndarray): Entries of the conductivity matrix: param[0] = :math:`\\kappa_{11}`, param[1] = :math:`\\kappa_{22}`, param[2] = :math:`\\kappa_{12}`
+
+        Returns:
+            np.ndarray: An array containing the solution of the anisotropic heat conduction equation, where the first two indices correspond to the x and y coordinates, respectively, and the third index corresponds to the time.
+        """
+        # set up physical properties
+        time_span = np.array([0, 0.1])
+        plate_length = np.array([1, 1])
+
+        # define the grid
+        num_grid_points = 20
+        dx = plate_length[0] / num_grid_points
+        dy = plate_length[1] / num_grid_points
+        safety_factor = 1.25
+        dt = min(dx, dy) ** 2 / (
+            safety_factor * 4 * np.max(self.PARAM_LIMITS[:, 1])
+        )
+        x = jnp.linspace(0, 1, num_grid_points)
+        y = jnp.linspace(0, 1, num_grid_points)
+        t = jnp.arange(time_span[0], time_span[1] + dt, dt)
+        t = t.at[-1].set(time_span[1])
+
+        # initial solution
+        u = jnp.empty((len(x), len(y), len(t)))
+
+        # define the initial condition
+        u_init = jnp.zeros((len(x), len(y)))
+        u = u.at[:, :, 0].set(u_init)
+
+        # define the boundary conditions
+        u_top = jnp.ones((len(x), len(t)))
+        u_bottom = jnp.zeros((len(x), len(t)))
+        u_left = jnp.ones((len(y), len(t)))
+        u_right = jnp.zeros((len(y), len(t)))
+
+        u = u.at[:, 0, :].set(u_bottom)
+        u = u.at[:, -1, :].set(u_top)
+        u = u.at[0, :, :].set(u_left)
+        u = u.at[-1, :, :].set(u_right)
+
+        # solve numerically
+        for n in range(0, len(t) - 1):
```

Comment:
I didn't check how large t is, but this is very likely killing the jit compilation.
Jit does something like unrolling the loop or at least tracing it through all the iterations.

Consider using `lax.scan` or lax.fori_loop / lax.while_loop.
If you'd like to have some support, message me, after Tuesday ;)

Reply:
Thanks, that's a great tipp! I believe I got it working, but broke something else in the process :D I'll let you know as soon as I've fixed that.